homeassistant/components/harmony/remote.py

```diff
@@ -166,12 +168,20 @@ async def async_added_to_hass(self):
         """Complete the initialization."""
         _LOGGER.debug("%s: Harmony Hub added", self._name)
         # Register the callbacks
-        self._client.callbacks = ClientCallbackType(
-            new_activity=self.new_activity,
-            config_updated=self.new_config,
-            connect=self.got_connected,
-            disconnect=self.got_disconnected,
-        )
+        if self._activity_notify:
+            self._client.callbacks = ClientCallbackType(
+                new_activity_starting=self.new_activity,
+                config_updated=self.new_config,
+                connect=self.got_connected,
+                disconnect=self.got_disconnected,
+            )
+        else:
+            self._client.callbacks = ClientCallbackType(
+                new_activity=self.new_activity,
+                config_updated=self.new_config,
+                connect=self.got_connected,
+                disconnect=self.got_disconnected,
+            )
```

Comment:
```suggestion
        callbacks = {
           config_updated: self.new_config,
           connect: self.got_connected
           disconnect: self.got_disconnected
        }
        if self._activity_notify:
            callbacks["new_activity_starting"] = self.new_activity
        else:
             callbacks["new_activity"] = self.new_activity
        self._client.callbacks = ClientCallbackType(**callbacks)
```

Doesn't save that many lines but maybe something the above?

Reply:
> Is there a branch for aioharmony 0.2.5 ?

Yeah: 

> Is there a branch for aioharmony 0.2.5 ?

Yep, still need to finish some more testing on it:
https://github.com/ehendrix23/aioharmony/archive/dev.zip

Reply:
Yep, looks cleaner. :-) Updated.